src/main/scala/sigmastate/interpreter/Interpreter.scala

```diff
@@ -233,11 +233,31 @@ trait Interpreter {
       BooleanConstant.fromBoolean(l.compareTo(r) < 0)
     case LE(BigIntConstant(l), BigIntConstant(r)) =>
       BooleanConstant.fromBoolean(l.compareTo(r) <= 0)
-      
-    case IsMember(tree: AvlTreeConstant, ByteArrayConstant(key), ByteArrayConstant(proof)) =>
-      val bv = tree.createVerifier(SerializedAdProof @@ proof)
-      val res = bv.performOneOperation(Lookup(ADKey @@ key))
-      BooleanConstant.fromBoolean(res.isSuccess) // TODO should we also check res.get.isDefined
+
+    /*
+        case IsMember(tree: AvlTreeConstant, ByteArrayConstant(key), ByteArrayConstant(proof)) =>
+          val bv = tree.createVerifier(SerializedAdProof @@ proof)
+          val res = bv.performOneOperation(Lookup(ADKey @@ key))
+          BooleanConstant.fromBoolean(res.isSuccess  && res.get.isDefined)
+    */
+
+    //TODO: simplify logic below
+    case i: IsMember
+      if i.tree.evaluated && i.key.isEvaluated && i.proof.isEvaluated =>
+
+      val keyBytes = i.key match {
+        case constant: Constant[SCollection[SByte.type]] => constant.value
+        case _ => i.key.items.map(_.asInstanceOf[EvaluatedValue[SByte.type]]).map(_.value.asInstanceOf[Byte]).toArray
+      }
```

Comment:
@kushti it is possible to use matchCase extension method
`val keyBytes = i.key.matchCase(cc => cc.value, c => c.value)`